cattle/plugins/docker/compute.py

```diff
@@ -174,16 +174,15 @@ def _id_filter(id, container):
         return id == container_id
 
     def get_container(self, client, instance):
-        try:
-            if instance.externalId is not None:
-                return self.get_container_by(client, lambda x: self._id_filter(
-                    instance.externalId, x))
-        except AttributeError:
-            pass
-
         name = '/{0}'.format(instance.uuid)
-        return self.get_container_by(client,
-                                     lambda x: self._name_filter(name, x))
+        container = self.get_container_by(client,
+                                          lambda x: self._name_filter(name, x))
+        if container:
+            return container
+
+        if hasattr(instance, 'externalId') and instance.externalId:
```

Comment:
object in python-agent from events are weird.  They are `JsonObject`s and don't really act right in terms of KeyError and AttributeError.  Are you sure this hasattr() check works right?